### ISSUE TITLE
Header Titles Not Appearing

### ISSUE THREAD

**Issue by dplattipswich** [akgulebubekir/Maui.DataGrid#133]
**Describe the bug**

On .NET MAUI 8, I can't get Header titles to show in the control. The data rows show up fine, but the header remains blank, no matter what I do. Screenshot attached. Worked fine on Xamarin. Any ideas?
![Screenshot_20231208-135837](https://github.com/akgulebubekir/Maui.DataGrid/assets/24593474/14ee18cb-492f-4d56-a901-f010983677b5)

Code is 

 <dg:DataGrid Grid.Row="3" x:Name="TrainGrid"    
      HeaderBordersVisible="True" 
      HeaderHeight="80"  
      RowHeight="30"                          
      FontSize="{x:Static local:App.AppFontSize}"
      FontFamily="{x:Static local:App.AppFontFamily}"
      BorderColor="Gray" 
      RefreshingEnabled="False" 
         >

     <dg:DataGrid.Columns >
         <dg:DataGridColumn Title="Train #" PropertyName="TrainNumber" Width="2*" VerticalContentAlignment="Center"/>
         <dg:DataGridColumn Title="Departs" PropertyName="Departure" Width="3*" StringFormat='{}{0:hh:mm tt}' />
         <dg:DataGridColumn Title="Arrives" PropertyName="Arrival" Width="3*" StringFormat='{}{0:hh:mm tt}' />
     </dg:DataGrid.Columns>
 </dg:DataGrid>

Screenshot attached

**Comment by dplattipswich** [comment]
Looking in with the debugger, I can see that the underlying header labels are being created, with the correct colors, but somehow they're not being displayed. Any ideas?

**Comment by symbiogenesis** [comment]
What happens if you remove all the styling? Remove the BorderColor, FontFamily, FontSize, and HeaderBordersVisible properties.

**Comment by dplattipswich** [comment]
Did that. Data in columns still being extracted correctly, but still no headers. Code: 

         <dg:DataGrid Grid.Row="3" x:Name="TrainGrid"    
             HeaderHeight="80"  
             RowHeight="30"                          
             >
  
             <dg:DataGrid.Columns >
                 <dg:DataGridColumn Title="Train #" PropertyName="TrainNumber" Width="2*"  />
                 <dg:DataGridColumn Title="Departs" PropertyName="Departure" Width="3*" StringFormat='{}{0:hh:mm tt}' />
                 <dg:DataGridColumn Title="Arrives" PropertyName="Arrival" Width="3*" StringFormat='{}{0:hh:mm tt}' />
             </dg:DataGrid.Columns>
         </dg:DataGrid>

![Screenshot_20231214-103034](https://github.com/akgulebubekir/Maui.DataGrid/assets/24593474/606655ef-7fc0-4533-8787-03e52ad5883b)

**Comment by dplattipswich** [comment]
A decent workaround is the Community License for Syncfusion's controls, which contains a datagrid. It's free to small outfits, similar to the free MS Community Edition of Visual Studio.

**Comment by symbiogenesis** [comment]
I ran your code on .NET 8 with Android and two properties in my App class:

```csharp
    public static double AppFontSize => 16;

    public static string AppFontFamily => "Arial";
```

Everything looked fine. Closing this.

**Comment by dplattipswich** [comment]
Edward, here's what happened. I do not know why, but it's a data point for
you.

 

I couldn't get the column headers to appear, so I switched to SyncFusion's
DataGrid. Unfortunately, that crashed while scrolling on the iPhone, but not
Android. Further investigation seemed to refer to a known bug that MS stated
that they weren't fixing any time soon. So I decided to try yours again. 

 

I got the same result: no column headers despite the Xaml, no matter what I
did. But in the sample code, I saw an example of creating columns
programmatically, through C# code, so I added one of those.

 

Believe it or not, when I added that one column through an explicit C# call,
not only did that column header show up, but the three that I had declared
in XAML also appeared. When I took out the C# call, they all disappeared.
So I switched all my column creation over to C#, and now all the headers
appear properly.  Go figure. 

 

I don't have any explanation as to why. I'd have to guess something
involving timing.  Hope this helps.

 

Best, DSP
